tests/unit/test_pandas_df_utils.py

```diff
@@ -140,5 +141,50 @@ def test_csv_to_libffm():
         assert line == '1 1:1:1 2:2:3 3:3:1.0 4:4:1\n'
 
 
+def test_negative_feedback_sampler():
+    df = pd.DataFrame({
+        'userID': [1, 2, 3, 4, 4, 5, 5, 5],
+        'itemID': [1, 2, 3, 1, 2, 1, 2, 3],
+        'rating': [5, 5, 5, 5, 5, 5, 5, 5]
```

Comment:
It may be a good idea to add a test without the 'rating' column, to ensure the functionality in both cases.

Reply:
Done